core/server/src/main/java/alluxio/master/block/BlockMaster.java

```diff
@@ -610,8 +609,12 @@ private void processWorkerRemovedBlocks(MasterWorkerInfo workerInfo,
     for (long removedBlockId : removedBlockIds) {
       MasterBlockInfo masterBlockInfo = mBlocks.get(removedBlockId);
       if (masterBlockInfo == null) {
-        LOG.warn("Worker {} removed block {} but block does not exist.", workerInfo.getId(),
-            removedBlockId);
+        LOG.warn("Worker {} informs the removed block {}, but block metadata does not exist"
+            + "on Master!", workerInfo.getId(), removedBlockId);
+        // TODO(pfxuan): [ALLUXIO-1804] should find a better way to handle the removed blocks.
```

Comment:
What were you imagining for a "better way"? It looks like removeBlock() is safe (and correct) to call here. If a worker removes a block id, that block id should be removed from the worker metadata, regardless of if the block metadata exists or not.